Fix #3251 expose REST DSL services to quarkus openapis (#3481)
apache/camel-quarkus: docs/modules/ROOT/pages/reference/extensions/openapi-java.adoc

Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.

> == Usage

Reviewer: You can move the usage content into `usage.adoc` and there's some [smart logic](https://camel.apache.org/camel-quarkus/2.7.x/contributor-guide/extension-documentation.html) in one of the maven plugins which will combine it into the extension docs.

Author: yeah, I create ```extensions/openapi-java/runtime/src/main/doc/usage.adoc```.

> You can use this extension to expose REST DSL services to quarkus openapi. With

Reviewer: ```suggestion
You can use this extension to expose REST DSL services to Quarkus OpenAPI. With
```

> the `quarkus-smallrye-openapi`, you can access them by `/q/openapi?format=json`.

Reviewer: ```suggestion
`quarkus-smallrye-openapi`, you can access them by `/q/openapi?format=json`.
```

Refer to the https://quarkus.io/guides/openapi-swaggerui[Quarkus OpenAPI guide] for further information.

> By default, this experimental feature disable. You can enable it by

Reviewer: ```suggestion
This is an experimental feature disable. You can enable it by
```

[source, properties]
----
quarkus.camel.openapi.expose.enabled=true
----

[WARNING]
====
It doesn't support the rests services used in `org.apache.camel.builder.LambdaRouteBuilder` right now.

> Also, it can not use CDI injection in the RouteBuilder `configure()` since we get the rest definitions at build time while the CDI is unavailable.

Reviewer: ```suggestion
Also, it can not use CDI injection in the RouteBuilder `configure()` since we get the rest definitions at build time while  CDI is unavailable.
```
